src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,88 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [newTitle, setNewTitle] = useState('');
+  const [newDescription, setNewDescrption] = useState('');
+  const [newImgUrl, setNewImgUrl] = useState('');
+  const [newImdbUrl, setNewImdbUrl] = useState('');
+  const [newImdbId, setNewImdbId] = useState('');
+
+  const isDisable = (!newTitle || !newImgUrl || !newImdbUrl || !newImdbId);
+
+  const addMovie = (e: React.MouseEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    onAdd({
+      title: newTitle,
+      description: newDescription,
+      imgUrl: newImgUrl,
+      imdbUrl: newImdbUrl,
+      imdbId: newImdbId,
+    });
+
+    setCount(value => value + 1);
+
+    setNewTitle('');
+    setNewDescrption('');
+    setNewImgUrl('');
+    setNewImdbUrl('');
+    setNewImdbId('');
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={addMovie}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={newTitle}
+        onChange={(value) => setNewTitle(value)}
```

Comment:
```suggestion
        onChange={setNewTitle}
```
You don't need to wrap your state handler. Just pass it directly. Fix across your code.